### ISSUE TITLE
does not work when it is a nested struct 

### ISSUE THREAD

**Issue by andyandhope** [fangq/jsonlab#95]
e.g., 

jsonmesh=struct('MeshNode',{"MeshNode",[0 0 0;1 0 0;0 1 0;1 1 0;0 0 1;1 0 1;0 1 1;1 1 1]},...
  'MeshElem',[1 2 4 8;1 3 4 8;1 2 6 8;1 5 6 8;1 5 7 8;1 3 7 8],...
  'MeshSurf',[1 2 4;1 2 6;1 3 4;1 3 7;1 5 6;1 5 7;...
  2 8 4;2 8 6;3 8 4;3 8 7;5 8 6;5 8 7],...
  'MeshCreator','FangQ','MeshTitle','T6 Cube',...
  'SpecialData',[nan, inf, -inf])

**Comment by fangq** [comment]
@andyandhope, can you rephrase the question title and body - what does not work? what codes do I need to reproduce what you see? what version of JSONLab, is it matlab or octave?

**Comment by andyandhope** [comment]
I am using the most recent version of JSONLab in matlab. 

i try to conver the 'nested' struct varible as json. It seems not work when the structure object is a nested struct.

for example:
jsonmesh = struct('MeshNode', {'MeshNote', [0,0,0; 1,0,0;1,1,0]})

I am not sure if i make myself clear.

**Comment by fangq** [comment]
@andyandhope, first of all, the issue you are reporting, if I understood you correctly, is not JSONLab's problem, but a general matlab utility question.

if you are trying to create a "nested struct", your above command is not going to do it. what you wrote is not a nested struct, but a struct array - if there is a cell element inside a struct(), struct will create an array of struct.

if you want it to be a nested struct, you should use struct() as the value instead of a cell array. see the difference below


```
>> jsonmesh = struct('MeshNode', {'MeshNote', [0,0,0; 1,0,0;1,1,0]})


jsonmesh = 

  1×2 struct array with fields:

    MeshNode

>> jsonmesh = struct('MeshNode', struct('MeshNote', [0,0,0; 1,0,0;1,1,0]))

jsonmesh = 

  struct with fields:

    MeshNode: [1×1 struct]
```

closing as it is irrelevant to JSONLab functions
